paddle/fluid/ir/transforms/pd_op_to_kernel_pass.cc

```diff
@@ -49,11 +49,116 @@ std::unordered_map<std::string, phi::DataType> Str2PhiDataType = {
     {"DataType::BOOL", phi::DataType::BOOL},
 };
 
+std::unordered_set<std::string> unchange_output_list = {"pd.feed_with_place",
+                                                        "builtin.combine",
+                                                        "builtin.slice",
+                                                        "pd.feed",
+                                                        "pd.fetch",
+                                                        "builtin.set_parameter",
+                                                        "builtin.get_parameter",
+                                                        "pd.shadow_output"};
+
+bool NeedFallBackCpu(const ir::Operation* op,
+                     const std::string& kernel_fn_name,
+                     phi::KernelKey kernel_key) {
+  if (unchange_output_list.count(op->name())) {
+    return false;
+  }
+  if (kernel_fn_name == "") {
+    return false;
+  }
+  if (phi::KernelFactory::Instance().HasKernel(kernel_fn_name, kernel_key)) {
+    return false;
+  }
+
+  if (kernel_key.backend() == phi::Backend::GPUDNN) {
+    kernel_key.set_backend(phi::Backend::GPU);
+
+    if (phi::KernelFactory::Instance().HasKernel(kernel_fn_name, kernel_key)) {
+      return false;
+    }
+  }
+
+  kernel_key.set_backend(phi::Backend::CPU);
+  if (phi::KernelFactory::Instance().HasKernel(kernel_fn_name, kernel_key)) {
+    return true;
+  }
+
+  return false;
+}
+
+bool NeedFallBackFromGPUDNN2GPU(ir::Operation* op,
+                                const phi::KernelKey kernel_key) {
+  // NOTE(phlrain): keep the same kernel select strategy with
+  // GetExepectKernelKey
+  if (op->name() == "pd.pool2d" || op->name() == "pd.pool2d_grad") {
+    if (kernel_key.backend() == phi::Backend::GPUDNN &&
+        (op->attributes().at("adaptive").dyn_cast<ir::BoolAttribute>().data() ==
+         true)) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+ir::OpResult AddPlaceTransferOp(ir::OpResult in,
+                                ir::Type out_type,
+                                const phi::Place& src_place,
+                                const phi::Place& tar_place,
```

Comment:
```suggestion
                                const phi::Place& dst_place,
```

一般 dst 与 src 对应，这里的tar是想表示target吧？

Reply:
done